src/__tests__/node_redis.spec.js

```diff
@@ -118,13 +119,75 @@ tape('NodeJS Redis', function (t) {
         assert.true(await client.track('nicolas@split.io', 'user', 'test.redis.event', 18), 'If the event was succesfully queued the promise will resolve to true');
         assert.false(await client.track(), 'If the event was NOT succesfully queued the promise will resolve to false');
 
+        await client.ready(); // promise already resolved
         await client.destroy();
 
         // close server connection
         server.close().then(assert.end);
       });
   });
 
+  t.test('Connection ready and timed out', assert => {
+    const readyTimeout = 0.1; // 100 millis
+    const configWithShortTimeout = { ...config, startup: { readyTimeout } };
+    const sdk = SplitFactory(configWithShortTimeout);
+    const client = sdk.client();
+
+    const start = Date.now();
+    let readyTimestamp;
+
+    client.getTreatment('UT_Segment_member', 'always-on').then(treatment => {
+      assert.equal(treatment, 'on', 'Evaluations using Redis storage should be correct and resolved once Redis connection is stablished');
+    });
+    client.track('nicolas@split.io', 'user', 'test.redis.event', 18).then(result => {
+      assert.true(result, 'If the event was succesfully queued the promise will resolve to true once Redis connection is stablished');
+    });
+
+    // SDK_READY_TIMED_OUT event must be emitted after 100 millis
+    client.on(client.Event.SDK_READY_TIMED_OUT, () => {
+      const delay = Date.now() - start;
+      assert.true(nearlyEqual(delay, readyTimeout * 1000), 'SDK_READY_TIMED_OUT event must be emitted after 100 millis');
+    });
+
+    // alse, ready promise must be rejected after 100 millis
+    client.ready().catch(() => {
+      const delay = Date.now() - start;
+      assert.true(nearlyEqual(delay, readyTimeout * 1000), 'Ready promise must be rejected after 100 millis');
+
+      // initialize server to emit SDK_READY
+      initializeRedisServer().then(async (server) => {
+        readyTimestamp = Date.now();
+
+        try {
+          await client.ready();
+          assert.fail('Ready promise keeps being rejected until SDK_READY is emitted');
+        } catch (error) {
+          assert.pass('Ready promise keeps being rejected until SDK_READY is emitted');
+        }
+
+        client.on(client.Event.SDK_READY, async () => {
```

Comment:
Wouldn't this subscription be better placed outside the server initialization promise callback?

I mean.. does it need to be sequential? It shouldn't matter when the server is created in the whole flow. We should test:
- Timeout when there is no server to connect to, regardless of the readyTimeout value.
- Timeout when there is a server to connect to but the readyTimeout is shorter than the time required to connect. This scenario will end up emitting a SDK_READY shortly after the timeout, and returning a .ready() promise resolved where it was rejected.
- Ready when there is nothing odd.

Reply:
thanks. added the second case by creating a second factory with a very short readyTimedout value